packages/common/src/workflow-definition-options.ts

```diff
@@ -0,0 +1,18 @@
+import { VersioningBehavior } from './worker-deployments';
+
+/**
+ * Options that can be used when defining a workflow via {@link setWorkflowOptions}.
+ */
+export interface WorkflowDefinitionOptions {
+  versioningBehavior?: VersioningBehavior;
+}
+
+type AsyncFunction<Args extends any[], ReturnType> = (...args: Args) => Promise<ReturnType>;
+export type WorkflowDefinitionOptionsOrGetter = WorkflowDefinitionOptions | (() => WorkflowDefinitionOptions);
+
+/**
+ * A workflow function that has been defined with options from {@link WorkflowDefinitionOptions}.
+ */
+export interface WorkflowFunctionWithOptions<Args extends any[], ReturnType> extends AsyncFunction<Args, ReturnType> {
```

Comment:
Does this have any value being exported publicly? If yes, then `AsyncFunction` should also be exported. If not, then please add `@internal` and `@hidden` (I will have to experiment a bit more to figure out how to reduce this to a single annotation, but in the mean time, let's use both).